lib/checkleakautovar.cpp

```diff
@@ -284,7 +284,12 @@ void CheckLeakAutoVar::checkScope(const Token * const startToken,
                                   std::set<int> notzero,
                                   nonneg int recursiveCount)
 {
-    if (++recursiveCount > 1000)    // maximum number of "else if ()"
+#if ASAN
+    static const nonneg int recursiveLimit = 300;
```

Comment:
Just curious: why did you make it `static`?

Reply:
Because its `const`.

Reply:
I guess static might cause overhead (>=C++11 due to threadsafety) but has no benefit for the const variable (https://softwareengineering.stackexchange.com/questions/350501/is-there-any-benefit-to-to-define-constant-local-variables-as-static-c)

Reply:
No, this uses [constant-initialization](https://en.cppreference.com/w/cpp/language/constant_initialization), so the initialization is done at compile-time. If the initialization was done at runtime then yes there would be extra threadsafety. However, this is not the case because we are performing constant-initialization. From [Static local variables](https://en.cppreference.com/w/cpp/language/storage_duration#Static_local_variables) section from cppreference:

> Variables declared at block scope with the specifier static have static storage duration but are initialized the first time control passes through their declaration (unless their initialization is zero- or constant-initialization, which can be performed before the block is first entered).